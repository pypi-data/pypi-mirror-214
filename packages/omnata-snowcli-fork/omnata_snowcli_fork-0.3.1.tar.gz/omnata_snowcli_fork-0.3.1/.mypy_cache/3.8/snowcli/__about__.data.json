{".class": "MypyFile", "_fullname": "snowcli.__about__", "future_import_flags": ["annotations"], "is_partial_stub_package": false, "is_stub": false, "names": {".class": "SymbolTable", "VERSION": {".class": "SymbolTableNode", "kind": "Gdef", "node": {".class": "Var", "flags": ["is_ready", "is_inferred", "has_explicit_value"], "fullname": "snowcli.__about__.VERSION", "name": "VERSION", "type": "builtins.str"}}, "__annotations__": {".class": "SymbolTableNode", "kind": "Gdef", "node": {".class": "Var", "flags": ["is_ready"], "fullname": "snowcli.__about__.__annotations__", "name": "__annotations__", "type": {".class": "Instance", "args": ["builtins.str", {".class": "AnyType", "missing_import_name": null, "source_any": null, "type_of_any": 6}], "type_ref": "builtins.dict"}}}, "__doc__": {".class": "SymbolTableNode", "kind": "Gdef", "node": {".class": "Var", "flags": ["is_ready"], "fullname": "snowcli.__about__.__doc__", "name": "__doc__", "type": "builtins.str"}}, "__file__": {".class": "SymbolTableNode", "kind": "Gdef", "node": {".class": "Var", "flags": ["is_ready"], "fullname": "snowcli.__about__.__file__", "name": "__file__", "type": "builtins.str"}}, "__name__": {".class": "SymbolTableNode", "kind": "Gdef", "node": {".class": "Var", "flags": ["is_ready"], "fullname": "snowcli.__about__.__name__", "name": "__name__", "type": "builtins.str"}}, "__package__": {".class": "SymbolTableNode", "kind": "Gdef", "node": {".class": "Var", "flags": ["is_ready"], "fullname": "snowcli.__about__.__package__", "name": "__package__", "type": "builtins.str"}}, "annotations": {".class": "SymbolTableNode", "cross_ref": "__future__.annotations", "kind": "Gdef"}}, "path": "src/snowcli/__about__.py"}