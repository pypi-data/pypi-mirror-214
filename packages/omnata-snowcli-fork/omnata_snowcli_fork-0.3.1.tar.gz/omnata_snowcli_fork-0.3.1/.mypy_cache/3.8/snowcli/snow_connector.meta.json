{"data_mtime": 1685973708, "dep_lines": [13, 1, 3, 4, 5, 6, 7, 10, 1, 1, 1, 1, 1, 1, 1, 1, 1, 11, 9, 9], "dep_prios": [5, 5, 10, 10, 5, 5, 5, 5, 5, 30, 30, 30, 30, 30, 30, 30, 30, 5, 10, 20], "dependencies": ["snowcli.snowsql_config", "__future__", "os", "pkgutil", "io", "pathlib", "typing", "jinja2", "builtins", "_collections_abc", "_typeshed", "abc", "jinja2.bccache", "jinja2.environment", "jinja2.loaders", "jinja2.runtime", "typing_extensions"], "hash": "4eb162e05830758bbc7b8b14f9258ab7206d78a1a96152584e1f9930d8c5a17b", "id": "snowcli.snow_connector", "ignore_all": false, "interface_hash": "de128d496c5f6b357b70a5317fdb404f847eaff8d4de31048e4529067aa6a8e6", "mtime": 1685963323, "options": {"allow_redefinition": false, "allow_untyped_globals": false, "always_false": [], "always_true": [], "bazel": false, "check_untyped_defs": false, "disable_bytearray_promotion": false, "disable_error_code": [], "disable_memoryview_promotion": false, "disabled_error_codes": [], "disallow_any_decorated": false, "disallow_any_explicit": false, "disallow_any_expr": false, "disallow_any_generics": false, "disallow_any_unimported": false, "disallow_incomplete_defs": false, "disallow_subclassing_any": false, "disallow_untyped_calls": false, "disallow_untyped_decorators": false, "disallow_untyped_defs": false, "enable_error_code": [], "enabled_error_codes": [], "follow_imports": "normal", "follow_imports_for_stubs": false, "ignore_errors": false, "ignore_missing_imports": true, "implicit_optional": false, "implicit_reexport": true, "local_partial_types": false, "mypyc": false, "platform": "linux", "plugins": [], "strict_concatenate": false, "strict_equality": false, "strict_optional": true, "warn_no_return": true, "warn_return_any": false, "warn_unreachable": false, "warn_unused_ignores": false}, "path": "src/snowcli/snow_connector.py", "plugin_data": null, "size": 17003, "suppressed": ["snowflake.connector.cursor", "snowflake.connector", "snowflake"], "version_id": "0.990"}