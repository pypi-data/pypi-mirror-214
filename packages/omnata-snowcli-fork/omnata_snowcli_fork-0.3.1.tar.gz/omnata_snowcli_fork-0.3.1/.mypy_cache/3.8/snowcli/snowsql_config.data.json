{".class": "MypyFile", "_fullname": "snowcli.snowsql_config", "future_import_flags": ["annotations"], "is_partial_stub_package": false, "is_stub": false, "names": {".class": "SymbolTable", "SnowsqlConfig": {".class": "SymbolTableNode", "kind": "Gdef", "node": {".class": "TypeInfo", "_promote": [], "abstract_attributes": [], "bases": ["builtins.object"], "declared_metaclass": null, "defn": {".class": "ClassDef", "fullname": "snowcli.snowsql_config.SnowsqlConfig", "name": "SnowsqlConfig", "type_vars": []}, "deletable_attributes": [], "flags": [], "fullname": "snowcli.snowsql_config.SnowsqlConfig", "has_param_spec_type": false, "metaclass_type": null, "metadata": {}, "module_name": "snowcli.snowsql_config", "mro": ["snowcli.snowsql_config.SnowsqlConfig", "builtins.object"], "names": {".class": "SymbolTable", "__init__": {".class": "SymbolTableNode", "kind": "Mdef", "node": {".class": "FuncDef", "abstract_status": 0, "arg_kinds": [0, 1], "arg_names": ["self", "path"], "flags": [], "fullname": "snowcli.snowsql_config.SnowsqlConfig.__init__", "name": "__init__", "type": null}}, "add_connection": {".class": "SymbolTableNode", "kind": "Mdef", "node": {".class": "FuncDef", "abstract_status": 0, "arg_kinds": [0, 0, 0], "arg_names": ["self", "connection_name", "entry"], "flags": [], "fullname": "snowcli.snowsql_config.SnowsqlConfig.add_connection", "name": "add_connection", "type": null}}, "config": {".class": "SymbolTableNode", "implicit": true, "kind": "Mdef", "node": {".class": "Var", "flags": ["is_inferred"], "fullname": "snowcli.snowsql_config.SnowsqlConfig.config", "name": "config", "type": {".class": "AnyType", "missing_import_name": null, "source_any": null, "type_of_any": 1}}}, "get_connection": {".class": "SymbolTableNode", "kind": "Mdef", "node": {".class": "FuncDef", "abstract_status": 0, "arg_kinds": [0, 0], "arg_names": ["self", "connection_name"], "flags": [], "fullname": "snowcli.snowsql_config.SnowsqlConfig.get_connection", "name": "get_connection", "type": null}}, "path": {".class": "SymbolTableNode", "implicit": true, "kind": "Mdef", "node": {".class": "Var", "flags": ["is_inferred"], "fullname": "snowcli.snowsql_config.SnowsqlConfig.path", "name": "path", "type": {".class": "AnyType", "missing_import_name": null, "source_any": null, "type_of_any": 1}}}}, "slots": null, "tuple_type": null, "type_vars": [], "typeddict_type": null}}, "__annotations__": {".class": "SymbolTableNode", "kind": "Gdef", "node": {".class": "Var", "flags": ["is_ready"], "fullname": "snowcli.snowsql_config.__annotations__", "name": "__annotations__", "type": {".class": "Instance", "args": ["builtins.str", {".class": "AnyType", "missing_import_name": null, "source_any": null, "type_of_any": 6}], "type_ref": "builtins.dict"}}}, "__doc__": {".class": "SymbolTableNode", "kind": "Gdef", "node": {".class": "Var", "flags": ["is_ready"], "fullname": "snowcli.snowsql_config.__doc__", "name": "__doc__", "type": "builtins.str"}}, "__file__": {".class": "SymbolTableNode", "kind": "Gdef", "node": {".class": "Var", "flags": ["is_ready"], "fullname": "snowcli.snowsql_config.__file__", "name": "__file__", "type": "builtins.str"}}, "__name__": {".class": "SymbolTableNode", "kind": "Gdef", "node": {".class": "Var", "flags": ["is_ready"], "fullname": "snowcli.snowsql_config.__name__", "name": "__name__", "type": "builtins.str"}}, "__package__": {".class": "SymbolTableNode", "kind": "Gdef", "node": {".class": "Var", "flags": ["is_ready"], "fullname": "snowcli.snowsql_config.__package__", "name": "__package__", "type": "builtins.str"}}, "annotations": {".class": "SymbolTableNode", "cross_ref": "__future__.annotations", "kind": "Gdef"}, "configparser": {".class": "SymbolTableNode", "cross_ref": "configparser", "kind": "Gdef"}, "os": {".class": "SymbolTableNode", "cross_ref": "os", "kind": "Gdef"}}, "path": "src/snowcli/snowsql_config.py"}