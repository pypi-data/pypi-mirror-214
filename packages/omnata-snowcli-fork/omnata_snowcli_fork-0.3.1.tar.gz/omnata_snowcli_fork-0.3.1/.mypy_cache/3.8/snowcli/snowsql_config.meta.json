{"data_mtime": 1685973708, "dep_lines": [1, 3, 4, 1, 1, 1, 1, 1], "dep_prios": [5, 10, 10, 5, 30, 30, 30, 30], "dependencies": ["__future__", "configparser", "os", "builtins", "abc", "io", "posixpath", "typing"], "hash": "463d7b80966725bffd76c441ae57d9a81d2aa7f67344f3ec0e3aa9a29fc84206", "id": "snowcli.snowsql_config", "ignore_all": false, "interface_hash": "1dc2a2acb4c8b01ca8b5e9a0dc74cbfc184951982e7a7e9a453d033b3c8078bb", "mtime": 1685963323, "options": {"allow_redefinition": false, "allow_untyped_globals": false, "always_false": [], "always_true": [], "bazel": false, "check_untyped_defs": false, "disable_bytearray_promotion": false, "disable_error_code": [], "disable_memoryview_promotion": false, "disabled_error_codes": [], "disallow_any_decorated": false, "disallow_any_explicit": false, "disallow_any_expr": false, "disallow_any_generics": false, "disallow_any_unimported": false, "disallow_incomplete_defs": false, "disallow_subclassing_any": false, "disallow_untyped_calls": false, "disallow_untyped_decorators": false, "disallow_untyped_defs": false, "enable_error_code": [], "enabled_error_codes": [], "follow_imports": "normal", "follow_imports_for_stubs": false, "ignore_errors": false, "ignore_missing_imports": true, "implicit_optional": false, "implicit_reexport": true, "local_partial_types": false, "mypyc": false, "platform": "linux", "plugins": [], "strict_concatenate": false, "strict_equality": false, "strict_optional": true, "warn_no_return": true, "warn_return_any": false, "warn_unreachable": false, "warn_unused_ignores": false}, "path": "src/snowcli/snowsql_config.py", "plugin_data": null, "size": 996, "suppressed": [], "version_id": "0.990"}