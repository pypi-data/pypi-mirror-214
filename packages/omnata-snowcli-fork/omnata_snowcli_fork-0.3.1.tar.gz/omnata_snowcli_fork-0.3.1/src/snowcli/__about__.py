from __future__ import annotations

VERSION = "0.3.1"
