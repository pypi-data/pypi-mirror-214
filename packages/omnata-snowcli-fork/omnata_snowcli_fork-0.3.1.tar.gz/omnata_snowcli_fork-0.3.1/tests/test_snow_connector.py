import textwrap
from unittest import mock

import pytest
from snowcli.snow_connector import SnowflakeConnector


# Used as a solution to syrupy having some problems with comparing multilines string
class CustomStr(str):
    def __repr__(self):
        return str(self)


@mock.patch("snowflake.connector")
def test_create_function(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.create_function(
        name="nameValue",
        input_parameters="(string a, variant b)",
        return_type="returnTypeValue",
        handler="handlerValue",
        imports="import1, import2",
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        overwrite=True,
        packages=["aaa", "bbb"],
    )
    query_io, *_ = connector.ctx.execute_stream.call_args.args
    query_str = query_io.getvalue()
    assert CustomStr(query_str) == snapshot


@mock.patch("snowflake.connector")
def test_create_procedure(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.create_procedure(
        name="nameValue",
        input_parameters="(string a, variant b)",
        return_type="returnTypeValue",
        handler="handlerValue",
        imports="import1, import2",
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        overwrite=True,
        packages=["aaa", "bbb"],
        execute_as_caller=True,
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_execute_function(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.execute_function(
        function="functionValue",
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_execute_procedure(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.execute_procedure(
        procedure="procedureValue",
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_describe_function(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.describe_function(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        signature="signatureValue",
        name="nameValue",
        input_parameters="(string a, variant b)",
        show_exceptions="show_exceptionsValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_describe_procedure(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.describe_procedure(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        signature="signatureValue",
        name="nameValue",
        input_parameters="(string a, variant b)",
        show_exceptions="show_exceptionsValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_list_functions(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.list_functions(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        like="likeValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_list_stages(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.list_stages(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        like="likeValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@pytest.mark.parametrize(
    "stage_name", [("namedStageValue"), ("snow://embeddedStageValue")]
)
@mock.patch("snowflake.connector")
def test_list_stage(_, snapshot, stage_name):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.list_stage(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        name=stage_name,
        like="likeValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@pytest.mark.parametrize(
    "stage_name", [("namedStageValue"), ("snow://embeddedStageValue")]
)
@mock.patch("snowflake.connector")
def test_get_stage(_, snapshot, stage_name):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.get_stage(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        name=stage_name,
        path="pathValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_set_procedure_comment(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.set_procedure_comment(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        signature="signatureValue",
        name="nameValue",
        input_parameters="(string a, variant b)",
        show_exceptions="show_exceptionsValue",
        comment="commentValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@pytest.mark.parametrize(
    "stage_name", [("namedStageValue"), ("snow://embeddedStageValue")]
)
@mock.patch("snowflake.connector")
def test_put_stage(_, snapshot, stage_name):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.put_stage(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        name=stage_name,
        path="pathValue",
        overwrite=True,
        parallel="parallelValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert textwrap.dedent(query.getvalue()) == snapshot


@pytest.mark.parametrize(
    "stage_name", [("namedStageValue"), ("snow://embeddedStageValue")]
)
@mock.patch("snowflake.connector")
def test_remove_from_stage(_, snapshot, stage_name):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.remove_from_stage(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        name=stage_name,
        path="/pathValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_create_stage(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.create_stage(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        name="nameValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_drop_stage(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.drop_stage(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        name="nameValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_list_procedures(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.list_procedures(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        like="likeValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_drop_function(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.drop_function(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        signature="signatureValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_drop_procedure(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.drop_procedure(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        signature="signatureValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_list_streamlits(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.list_streamlits(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_show_warehouses(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.show_warehouses(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        like="likeValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_create_streamlit(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.create_streamlit(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        name="nameValue",
        file="fileValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_create_streamlit_from_stage(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.create_streamlit(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        name="nameValue",
        file="fileValue",
        from_stage_command="FROM @stageValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_share_streamlit(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.share_streamlit(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        name="nameValue",
        to_role="to_roleValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_drop_streamlit(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.drop_streamlit(
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
        name="nameValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_deploy_streamlit(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (mock.MagicMock(),)
    connector.upload_file_to_stage = mock.MagicMock()

    connector.deploy_streamlit(
        name="nameValue",
        file_path="file_pathValue",
        stage_path="stage_pathValue",
        role="roleValue",
        database="databaseValue",
        schema="schemaValue",
        overwrite=True,
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot


@mock.patch("snowflake.connector")
def test_describe_streamlit(_, snapshot):
    connector = SnowflakeConnector(
        connection_name="foo", snowsql_config=mock.MagicMock()
    )
    connector.ctx.execute_stream.return_value = (None, None)

    connector.describe_streamlit(
        name="nameValue",
        database="databaseValue",
        schema="schemaValue",
        role="roleValue",
        warehouse="warehouseValue",
    )
    query, *_ = connector.ctx.execute_stream.call_args.args
    assert query.getvalue() == snapshot
