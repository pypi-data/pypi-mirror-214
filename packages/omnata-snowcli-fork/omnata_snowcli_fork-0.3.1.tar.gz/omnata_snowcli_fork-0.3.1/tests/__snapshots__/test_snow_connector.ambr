# serializer version: 1
# name: test_create_function
  '''
  use database databaseValue;
  use schema schemaValue;
  use warehouse warehouseValue;
  CREATE OR REPLACE  FUNCTION nameValue(string a, variant b)
           RETURNS returnTypeValue
           LANGUAGE PYTHON
           RUNTIME_VERSION=3.8
           IMPORTS=('import1, import2')
           HANDLER='handlerValue'
           PACKAGES=('aaa','bbb');
  describe function nameValue(a, b);
  '''
# ---
# name: test_create_procedure
  '''
  use database databaseValue;
  use schema schemaValue;
  use warehouse warehouseValue;
  
  CREATE OR REPLACE  PROCEDURE nameValue(string a, variant b)
           RETURNS returnTypeValue
           LANGUAGE PYTHON
           RUNTIME_VERSION=3.8
           IMPORTS=('import1, import2')
           HANDLER='handlerValue'
           PACKAGES=('aaa','bbb')
           EXECUTE AS CALLER;
  
  
  describe PROCEDURE nameValue(a, b);
  '''
# ---
# name: test_create_stage
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  
  create stage if not exists nameValue;
  '''
# ---
# name: test_create_streamlit
  '''
  use role roleValue;
  use database databaseValue;
  use schema schemaValue;
  use warehouse warehouseValue;
  
  create streamlit nameValue
    
    MAIN_FILE = 'fileValue'
    QUERY_WAREHOUSE = warehouseValue;
  
  show streamlits;
  describe streamlit nameValue;
  
  alter streamlit nameValue checkout;
  '''
# ---
# name: test_create_streamlit_from_stage
  '''
  use role roleValue;
  use database databaseValue;
  use schema schemaValue;
  use warehouse warehouseValue;
  
  create streamlit nameValue
    FROM @stageValue
    MAIN_FILE = 'fileValue'
    QUERY_WAREHOUSE = warehouseValue;
  
  show streamlits;
  describe streamlit nameValue;
  
  alter streamlit nameValue checkout;
  '''
# ---
# name: test_deploy_streamlit
  '''
  use database databaseValue;
  use schema schemaValue;
  use role roleValue;
  
  CALL SYSTEM$GENERATE_STREAMLIT_URL_FROM_NAME('nameValue');
  '''
# ---
# name: test_describe_function
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  desc FUNCTION signatureValue;
  '''
# ---
# name: test_describe_procedure
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  desc PROCEDURE signatureValue;
  '''
# ---
# name: test_describe_streamlit
  '''
  use database databaseValue;
  use schema schemaValue;
  use role roleValue;
  
  CALL SYSTEM$GENERATE_STREAMLIT_URL_FROM_NAME('nameValue');
  '''
# ---
# name: test_drop_function
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  drop function signatureValue;
  '''
# ---
# name: test_drop_procedure
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  drop procedure signatureValue;
  '''
# ---
# name: test_drop_stage
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  
  drop stage nameValue;
  '''
# ---
# name: test_drop_streamlit
  '''
  use role roleValue;
  use database databaseValue;
  use schema schemaValue;
  use warehouse warehouseValue;
  drop streamlit "nameValue";
  '''
# ---
# name: test_execute_function
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  select functionValue;
  '''
# ---
# name: test_execute_procedure
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  call procedureValue;
  '''
# ---
# name: test_get_stage[namedStageValue]
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  
  get @namedStageValue file://pathValue/
  '''
# ---
# name: test_get_stage[snow://embeddedStageValue]
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  
  get snow://embeddedStageValue file://pathValue/
  '''
# ---
# name: test_list_functions
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  show user functions like 'likeValue';
  select "name", "created_on", "arguments", "language" from table(result_scan(last_query_id()));
  '''
# ---
# name: test_list_procedures
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  show user procedures like 'likeValue';
  select "name", "created_on", "arguments" from table(result_scan(last_query_id()));
  '''
# ---
# name: test_list_stage[namedStageValue]
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  ls @namedStageValue;
  '''
# ---
# name: test_list_stage[snow://embeddedStageValue]
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  ls snow://embeddedStageValue;
  '''
# ---
# name: test_list_stages
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  show stages;
  '''
# ---
# name: test_list_streamlits
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  show streamlits;
  '''
# ---
# name: test_put_stage[namedStageValue]
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  
  put file://pathValue @namedStageValue auto_compress=false parallel=parallelValue overwrite=True;
  '''
# ---
# name: test_put_stage[snow://embeddedStageValue]
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  
  put file://pathValue snow://embeddedStageValue auto_compress=false parallel=parallelValue overwrite=True;
  '''
# ---
# name: test_remove_from_stage[namedStageValue]
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  
  remove @namedStageValue/pathValue
  '''
# ---
# name: test_remove_from_stage[snow://embeddedStageValue]
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  
  remove snow://embeddedStageValue/pathValue
  '''
# ---
# name: test_set_procedure_comment
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  alter PROCEDURE signatureValue SET COMMENT = $$commentValue$$;
  '''
# ---
# name: test_share_streamlit
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  
  grant usage on streamlit nameValue to role to_roleValue;
  '''
# ---
# name: test_show_warehouses
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  show warehouses;
  '''
# ---
